Tests/UpstreamIntegrationTests/UpstreamIntegrationTests.swift

```diff
@@ -117,6 +116,282 @@ class UpstreamIntegrationTests: TestBase {
+
+        // Verify
+        // MARK: Network response assertions
+        let matchedResponse = networkService.getResponsesFor(networkRequest: locationHintNetworkRequest, timeout: 5)
+        XCTAssertEqual(200, matchedResponse.first?.responseCode)
+
+        // MARK: Response Event assertions
+        // Only validate for the location hint relevant to Edge Network extension
+        let expectedLocationHintJSON = #"""
+        {
+          "payload": [
+            {
+              "ttlSeconds" : 1800,
+              "scope" : "EdgeNetwork",
+              "hint" : "va6"
+            }
+          ]
+        }
+        """#
+
+        assertEdgeResponseEvent(expectedJSON: expectedLocationHintJSON, eventSource: TestConstants.EventSource.LOCATION_HINT_RESULT, exactMatchPaths: ["payload[*].scope", "payload[*].hint"])
+
+        let expectedStateStore1stJSON = #"""
+        {
+          "payload": [
+            {
+              "maxAge": 1,
+              "key": "stringType",
+              "value": "stringType"
+            },
+            {
+              "maxAge": 1,
+              "key": "stringType",
+              "value": "stringType"
+            }
+          ]
+        }
+        """#
+
+        assertEdgeResponseEvent(expectedJSON: expectedStateStore1stJSON, eventSource: TestConstants.EventSource.STATE_STORE)
+    }
+
+    // MARK: - Error scenarios
+
+    // Tests that an invalid datastream ID returns the expected error
+    func testSendEvent_withInvalidDatastreamID_receivesExpectedError() {
+        // Setup
+        let standardNetworkRequest = NetworkRequest(urlString: "https://obumobile5.data.adobedc.net/ee/v1/interact", httpMethod: .post)!
+
+        networkService.setExpectationForNetworkRequest(networkRequest: standardNetworkRequest, expectedCount: 1)
+
+        MobileCore.updateConfigurationWith(configDict: ["edge.configId": "12345-example"])
+        // Test
+        let experienceEvent = ExperienceEvent(xdm: ["xdmtest": "data"],
+                                              data: ["data": ["test": "data"]])
+        Edge.sendEvent(experienceEvent: experienceEvent)
+
+        // Verify
+        // MARK: Network response assertions
+        let matchedResponse = networkService.getResponsesFor(networkRequest: standardNetworkRequest, timeout: 5)
+        XCTAssertEqual(400, matchedResponse.first?.responseCode)
+
+        // MARK: Event assertions
+        let expectedErrorJSON = #"""
+        {
+            "status": 400,
+            "detail": "stringType",
+            "report": {
+              "requestId": "stringType"
+            },
+            "requestEventId": "stringType",
+            "title": "Invalid datastream ID",
+            "type": "stringType",
+            "requestId": "stringType"
+          }
+        """#
+
+        assertEdgeResponseEvent(expectedJSON: expectedErrorJSON, eventSource: TestConstants.EventSource.ERROR_RESPONSE_CONTENT, exactMatchPaths: ["status", "title"])
```

Comment:
can we assert on the Edge error code too?

Reply:
the error code seems to be under type, e.g. `"type" : "https://ns.adobe.com/aep/errors/EXEG-0104-422"`

Reply:
thanks. that is ok, that is the namespace URI and it follows the format: "A URI reference (RFC3986) that identifies the problem type, following the format https://ns.adobe.com/aep/errors/<ERROR-CODE>"